airbyte-ci/connectors/pipelines/pipelines/actions/environments.py

```diff
@@ -82,6 +82,7 @@ def with_testing_dependencies(context: PipelineContext) -> Container:
 
     return (
         python_environment.with_exec(["pip", "install"] + CONNECTOR_TESTING_REQUIREMENTS)
+        .with_exec(sh_dash_c(["curl -fsSL https://get.docker.com | sh"]))
```

Comment:
python test containers now include docker

Reply:
So the use case in here is we want to be able to use docker in tests to spin up containers nessessary for certain test environments. e.g. source-file needs docker compose for its test "remote" file server

(@pedroslopez do I have that right?)

In any sense looking at the code we already have I see `with_global_dockerd_service` and `with_docker_cli` inside the `environments.py` mega file.

But I dont think those are useful here for installing docker-compose. 

(@alafanechere can you fact check me here?)

So if thats all true then I think were on the right track but should have a modification
1. Lets not use curl
1. Lets install docker / docker-compose via apt-get

Reply:
FYI: In https://github.com/airbytehq/airbyte/pull/30474 this `with_testing_dependencies` function is removed. Connectors test will run inside the connector container on top of which we'd install its testing dependencies according to its setup.py.
I believe docker as a system dependency should not be added to this function, nor to a future "pre/post" hook for source-file. 

If we need docker-compose for testing I suggest to install the pip distributed docker-compose by declaring `pytest-docker[docker-compose-v1]` as a test requirement. see https://github.com/avast/pytest-docker .

If we really want to use the latest docker-compose version I suggest to only [install the docker-compose plugin](https://docs.docker.com/compose/install/linux/#install-the-plugin-manually). This can be added to `with_testing_dependencies` in this PR but eventually added to a pre/post build hook following https://github.com/airbytehq/airbyte/pull/30474 .

I'd prefer that we stick to the pip distributed docker compose as it will not require a pre/post build hook implementation for this connector. And I believe it's fine to use an old docker compose version in this context.